ledger/participant-state/kvutils/tools/integrity-check/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/LogEntryIdComputationStrategy.scala

```diff
@@ -0,0 +1,38 @@
+package com.daml.ledger.participant.state.kvutils.tools.integritycheck
+
+import java.security.MessageDigest
+
+import com.daml.ledger.participant.state.kvutils.Raw
+import com.google.protobuf.ByteString
+
+trait LogEntryIdComputationStrategy {
+  def compute(key: Raw.Key, value: Raw.Envelope): Raw.LogEntryId
+}
+
+final class RawLogEntryIdComputationStrategy extends LogEntryIdComputationStrategy {
+  override def compute(key: Raw.Key, value: Raw.Envelope): Raw.LogEntryId =
+    Raw.LogEntryId(key.bytes) // `key` is of an unknown type.
+}
+
+final class HmmLogEntryIdComputationStrategy(writeItemParser: WriteItemParser)
+    extends LogEntryIdComputationStrategy {
+
+  private val LogEntryIdPrefix = "0"
+
+  override def compute(key: Raw.Key, value: Raw.Envelope): Raw.LogEntryId =
+    writeItemParser.parse((key, value)) match {
+      case Right(Left((_, logEntry))) =>
+        throw new RuntimeException(s"compute: log entry ($logEntry)") // FIXME
+      case Right(Right((stateKey, _))) =>
+        val messageDigest = MessageDigest
+          .getInstance("SHA-256")
+        messageDigest.update(stateKey.toByteArray)
```

Comment:
Ignore this so far, as this has a similar (wrong) result to using just `Raw.Key`.